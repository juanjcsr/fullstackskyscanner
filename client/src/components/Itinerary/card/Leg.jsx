import React from 'react';

import { BpkGridContainer, BpkGridRow, BpkGridColumn } from 'bpk-component-grid';
import BpkText from 'bpk-component-text';
import BpkSmallLongArrowRigthIcon  from 'bpk-component-icon/sm/long-arrow-right';
import BpkSmallLongArrowLeftIcon  from 'bpk-component-icon/sm/long-arrow-left';
import AirlineIcon from '../card/Icon';

const datestringToDate = ( dateString ) =>{
    const d = new Date(dateString);
    return `${d.getHours()}:${d.getMinutes()}`
}


const Leg = ({legData, directionality, ...props}) => {
    const legTime = `${Math.floor(legData.Duration / 60)}h ${legData.Duration % 60}m `;
    const stops = legData.Stops.length === 0 ? "Direct" : `${legData.Stops.length} Stops`;
    const firstCarrier = legData.SegmentsDetail[0].CarrierName[0];
    const color = stops === 'Direct' ? 'rgb(0, 215, 117)' : 'rgb(255, 84, 82)';
    const backgroundColor = {
        "color": color
    };
    return (        
        <>
            <BpkGridColumn width={1}>
                <AirlineIcon iconData={firstCarrier.Code}></AirlineIcon>
            </BpkGridColumn>
            <BpkGridColumn width={2}>
                <BpkText tagName="p">{datestringToDate(legData.Departure)}</BpkText>
                <BpkText tagName="p">{legData.OriginStationPlace[0].Code}</BpkText>
            </BpkGridColumn>
            <BpkGridColumn width={1}>
                {directionality === "Outbound" ? <BpkSmallLongArrowRigthIcon /> :<BpkSmallLongArrowLeftIcon />}
            </BpkGridColumn>
            <BpkGridColumn width={2}>
                <BpkText tagName="p">{datestringToDate(legData.Arrival)}</BpkText>
                <BpkText tagName="p">{legData.DestinationStationPlace[0].Code}</BpkText>
            </BpkGridColumn>   
            <BpkGridColumn width={3} offset={3}>
                <BpkText tagName="p">{legTime}</BpkText>
                <BpkText tagName="p" style={backgroundColor}>{stops}</BpkText>
            </BpkGridColumn>
        </>
    )
}

export default Leg;