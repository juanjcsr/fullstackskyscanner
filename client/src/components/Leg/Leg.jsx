import React from 'react';
import PropTypes from 'prop-types';

import { BpkGridColumn } from 'bpk-component-grid';
import BpkText from 'bpk-component-text';
import BpkSmallLongArrowRigthIcon from 'bpk-component-icon/sm/long-arrow-right';
import BpkSmallLongArrowLeftIcon from 'bpk-component-icon/sm/long-arrow-left';
import AirlineIcon from './../AirlineIcon/AirlineIcon';

const datestringToDate = (dateString) => {
  const d = new Date(dateString);
  return `${d.getHours()}:${d.getMinutes()}`;
};

const Leg = ({ legData, directionality }) => {
  const legTime = `${Math.floor(legData.Duration / 60)}h ${legData.Duration % 60}m `;
  const stops = legData.Stops.length === 0 ? 'Direct' : `${legData.Stops.length} Stops`;
  const firstCarrier = legData.SegmentsDetail[0].CarrierName[0];
  const color = stops === 'Direct' ? 'rgb(0, 215, 117)' : 'rgb(255, 84, 82)';
  const backgroundColor = {
    color,
  };
  return (
    <div>
      <BpkGridColumn width={1}>
        <AirlineIcon iconData={firstCarrier.Code} />
      </BpkGridColumn>
      <BpkGridColumn width={2}>
        <BpkText tagName="p">{datestringToDate(legData.Departure)}</BpkText>
        <BpkText tagName="p">{legData.OriginStationPlace[0].Code}</BpkText>
      </BpkGridColumn>
      <BpkGridColumn width={1}>
        {directionality === 'Outbound' ? <BpkSmallLongArrowRigthIcon /> : <BpkSmallLongArrowLeftIcon />}
      </BpkGridColumn>
      <BpkGridColumn width={2}>
        <BpkText tagName="p">{datestringToDate(legData.Arrival)}</BpkText>
        <BpkText tagName="p">{legData.DestinationStationPlace[0].Code}</BpkText>
      </BpkGridColumn>
      <BpkGridColumn width={3} offset={3}>
        <BpkText tagName="p">{legTime}</BpkText>
        <BpkText tagName="p" style={backgroundColor}>{stops}</BpkText>
      </BpkGridColumn>
    </div>
  );
};

Leg.propTypes = {
  legData: PropTypes.objectOf(PropTypes.any),
  directionality: PropTypes.string,
};

Leg.defaultProps = {
  legData: {},
  directionality: '',
};


export default Leg;
