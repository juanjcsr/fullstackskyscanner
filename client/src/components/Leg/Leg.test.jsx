import React from 'react';
import TestRenderer from 'react-test-renderer';

import Leg from './Leg';

describe('Leg', () => {
  it('should be rendered without data', () => {
    const component = TestRenderer.create(<Leg />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should be rendered correctly with data', () => {
    const legData = {
      Id: '11235-1903251950--32480-0-13554-1903252110',
      SegmentIds: [0],
      OriginStation: 11235,
      DestinationStation: 13554,
      Departure: '2019-03-25T19:50:00',
      Arrival: '2019-03-25T21:10:00',
      Duration: 80,
      JourneyMode: 'Flight',
      Stops: [],
      Carriers: [881],
      OperatingCarriers: [881],
      Directionality: 'Outbound',
      FlightNumbers: [{ FlightNumber: '1463', CarrierId: 881 }],
      SegmentsDetail: [{
        OriginStationPlace: [{
          Id: 11235, ParentId: 2343, Code: 'EDI', Type: 'Airport', Name: 'Edinburgh',
        }],
        DestinationStationPlace: [{
          Id: 13554, ParentId: 4698, Code: 'LHR', Type: 'Airport', Name: 'London Heathrow',
        }],
        CarrierName: [{
          Id: 881, Code: 'BA', Name: 'British Airways', ImageUrl: 'https://s1.apideeplink.com/images/airlines/BA.png', DisplayCode: 'BA',
        }],
        OperatingCarrier: 881,
        Id: 0,
        OriginStation: 11235,
        DestinationStation: 13554,
        DepartureDateTime: '2019-03-25T19:50:00',
        ArrivalDateTime: '2019-03-25T21:10:00',
        Carrier: 881,
        Duration: 80,
        FlightNumber: '1463',
        JourneyMode: 'Flight',
        Directionality: 'Outbound',
      }],
      OriginStationPlace: [{
        Id: 11235, ParentId: 2343, Code: 'EDI', Type: 'Airport', Name: 'Edinburgh',
      }],
      DestinationStationPlace: [{
        Id: 13554, ParentId: 4698, Code: 'LHR', Type: 'Airport', Name: 'London Heathrow',
      }],
    };
    const component = TestRenderer.create(<Leg legData={legData} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
