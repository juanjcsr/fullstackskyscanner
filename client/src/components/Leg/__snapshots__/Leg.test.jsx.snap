// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leg should be rendered correctly with data 1`] = `
<div>
  <div
    className="bpk-grid__column bpk-grid__column--1 bpk-grid__column--padded"
  >
    <div>
      <img
        alt=""
        src="https://logos.skyscnr.com/images/airlines/favicon/BA.png"
        style={
          Object {
            "position": "relative",
            "width": "30px",
          }
        }
      />
    </div>
  </div>
  <div
    className="bpk-grid__column bpk-grid__column--2 bpk-grid__column--padded"
  >
    <p
      className="bpk-text bpk-text--base"
    >
      19:50
    </p>
    <p
      className="bpk-text bpk-text--base"
    >
      EDI
    </p>
  </div>
  <div
    className="bpk-grid__column bpk-grid__column--1 bpk-grid__column--padded"
  >
    <svg
      height="18"
      style={
        Object {
          "height": "1.125rem",
          "width": "1.125rem",
        }
      }
      viewBox="0 0 24 24"
      width="18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9.6 4.5L3.9 9.9c-.4.4-.7.9-.8 1.5-.1.2-.1.4-.1.6 0 .2 0 .4.1.6.1.6.4 1.1.8 1.5l5.7 5.3c.8.8 2.1.7 2.8-.1.8-.8.7-2.1-.1-2.8L9.6 14H19c1.1 0 2-.9 2-2s-.9-2-2-2H9.6l2.7-2.5c.5-.4.7-1 .7-1.5s-.2-1-.5-1.4c-.8-.8-2.1-.8-2.9-.1z"
      />
    </svg>
  </div>
  <div
    className="bpk-grid__column bpk-grid__column--2 bpk-grid__column--padded"
  >
    <p
      className="bpk-text bpk-text--base"
    >
      21:10
    </p>
    <p
      className="bpk-text bpk-text--base"
    >
      LHR
    </p>
  </div>
  <div
    className="bpk-grid__column bpk-grid__column--3 bpk-grid__column--offset-3 bpk-grid__column--padded"
  >
    <p
      className="bpk-text bpk-text--base"
    >
      1h 20m 
    </p>
    <p
      className="bpk-text bpk-text--base"
      style={
        Object {
          "color": "rgb(0, 215, 117)",
        }
      }
    >
      Direct
    </p>
  </div>
</div>
`;

exports[`Leg should be rendered without data 1`] = `
<div>
  <div
    className="bpk-grid__column bpk-grid__column--1 bpk-grid__column--padded"
  >
    <div>
      <img
        alt=""
        src="https://logos.skyscnr.com/images/airlines/favicon/.png"
        style={
          Object {
            "position": "relative",
            "width": "30px",
          }
        }
      />
    </div>
  </div>
  <div
    className="bpk-grid__column bpk-grid__column--2 bpk-grid__column--padded"
  >
    <p
      className="bpk-text bpk-text--base"
    >
      NaN:NaN
    </p>
    <p
      className="bpk-text bpk-text--base"
    >
      
    </p>
  </div>
  <div
    className="bpk-grid__column bpk-grid__column--1 bpk-grid__column--padded"
  >
    <svg
      height="18"
      style={
        Object {
          "height": "1.125rem",
          "width": "1.125rem",
        }
      }
      viewBox="0 0 24 24"
      width="18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9.6 4.5L3.9 9.9c-.4.4-.7.9-.8 1.5-.1.2-.1.4-.1.6 0 .2 0 .4.1.6.1.6.4 1.1.8 1.5l5.7 5.3c.8.8 2.1.7 2.8-.1.8-.8.7-2.1-.1-2.8L9.6 14H19c1.1 0 2-.9 2-2s-.9-2-2-2H9.6l2.7-2.5c.5-.4.7-1 .7-1.5s-.2-1-.5-1.4c-.8-.8-2.1-.8-2.9-.1z"
      />
    </svg>
  </div>
  <div
    className="bpk-grid__column bpk-grid__column--2 bpk-grid__column--padded"
  >
    <p
      className="bpk-text bpk-text--base"
    >
      NaN:NaN
    </p>
    <p
      className="bpk-text bpk-text--base"
    >
      
    </p>
  </div>
  <div
    className="bpk-grid__column bpk-grid__column--3 bpk-grid__column--offset-3 bpk-grid__column--padded"
  >
    <p
      className="bpk-text bpk-text--base"
    >
      NaNh NaNm 
    </p>
    <p
      className="bpk-text bpk-text--base"
      style={
        Object {
          "color": "rgb(0, 215, 117)",
        }
      }
    >
      Direct
    </p>
  </div>
</div>
`;
